libvast/src/value_index.cpp

```diff
@@ -213,6 +213,55 @@ string_index::lookup_impl(relational_operator op, data_view x) const {
   ), x);
 }
 
+// -- enumeration_index --------------------------------------------------------
+
+enumeration_index::enumeration_index(vast::type t)
+  : value_index{std::move(t)},
+    index_{std::numeric_limits<enumeration>::max() + 1} {
+  // nop
+}
+
+caf::error enumeration_index::serialize(caf::serializer& sink) const {
+  return caf::error::eval([&] { return value_index::serialize(sink); },
+                          [&] { return sink(index_); });
+}
+
+caf::error enumeration_index::deserialize(caf::deserializer& source) {
+  return caf::error::eval([&] { return value_index::deserialize(source); },
+                          [&] { return source(index_); });
+}
+
+bool enumeration_index::append_impl(data_view x, id pos) {
+  if (auto e = caf::get_if<view<enumeration>>(&x)) {
+    index_.skip(pos - index_.size());
+    index_.append(*e);
+    return true;
+  }
+  return false;
+}
+
+expected<ids> enumeration_index::lookup_impl(relational_operator op,
+                                             data_view d) const {
+  if (offset() == 0) // FIXME: why do we need this check again?
```

Comment:
It's really just an optimization. If there's nothing in the index, there's nothing to return. to keep the logic as simple as possible, I'd remove this statement. It's premature anyway.